### ISSUE TITLE
Inter-namespace CiliumNetworkPolicy

### ISSUE THREAD

**Issue by michi-covalent** [cilium/cilium#966]
Right now CiliumNetworkPolicy gets created inside a kubernetes namespace, and you cannot override source and destination endpoint. It would be nice to be able to create a policy where source and destination are in different namespaces. Ideally I'd like to be able to do something like this:

```
kind: CiliumNetworkPolicy
apiVersion: cilium.io/v1
metadata:
  name: my-policy
spec:
  endpointSelector:
    matchLabels:
      "k8s:io.kubernetes.pod.namespace": "n1"
  ingress:
    - fromEndpoints:
        - matchLabels:
          "k8s:io.kubernetes.pod.namespace": "n2"
      toPorts:
        - ports:
            - port: '80'
              protocol: TCP
          rules: {}
```

**Comment by tgraf** [comment]
@aanm Can you summarize what is still missing to support this?

**Comment by aanm** [comment]
@michi-covalent to enforce the policies on a different namespace one should specify the namespace in the policy. This was implemented in the PR #1103 so make sure you're running the latest cilium image.

For example, the following CNP
```
metadata:
  name: "l7-stresstest"
  namespace: "development"
spec:
  endpointSelector:
    matchLabels:
      id: server
  ingress:
  - fromEndpoints:
    - matchLabels:
        id: client
        "k8s:io.kubernetes.pod.namespace": qa
```

Will be translated into:
```
"endpointSelector": {
  "matchLabels": {
    "any:id": "server",
    "k8s:io.kubernetes.pod.namespace": "development"
  }
},
"ingress": [
  {
    "fromEndpoints": [
      {
        "matchLabels": {
          "any:id": "client",
          "k8s:io.kubernetes.pod.namespace": "qa"
        }
      }
    ],
```

**Comment by michi-covalent** [comment]
@aanm hmm somehow that's not working for me:

```
 % cat minikube/resources/policy.yaml             
kind: CiliumNetworkPolicy
apiVersion: cilium.io/v2
metadata:
  name: mypolicy
  namespace: default
spec:
  endpointSelector:
    matchLabels:
      'k8s:io.kubernetes.pod.namespace': default
  ingress:
  - fromEndpoints:
    - matchLabels:
      'k8s:io.kubernetes.pod.namespace': kube-system
    toPorts:
    - ports:
      - port: "8080"
        protocol: TCP

root@minikube:/# cilium policy get
[
  {
    "endpointSelector": {
      "matchLabels": {
        "k8s:io.kubernetes.pod.namespace": "default"
      }
    },
    "ingress": [
      {
        "fromEndpoints": [
          {
            "matchLabels": {
              "k8s:io.kubernetes.pod.namespace": "default"
            }
          }
        ],
        "toPorts": [
          {
            "ports": [
              {
                "port": "8080",
                "protocol": "TCP"
              }
            ]
          }
        ]
      }
    ],
    "labels": [
      {
        "key": "io.cilium.k8s-policy-name",
        "value": "mypolicy",
        "source": "unspec"
      }
    ]
  }
]
Revision: 67
root@minikube:/# cilium version
Cilium 0.11.90 56548f8 Mon, 31 Jul 2017 15:28:23 -0700 go version go1.8.3 linux/amd64
```

**Comment by michi-covalent** [comment]
the issue was with the yaml file. the matchLabels section had wrong indentation:
```
kind: CiliumNetworkPolicy
apiVersion: cilium.io/v2
metadata:
  name: mypolicy
  namespace: default
spec:
  endpointSelector:
    matchLabels:
      'k8s:io.kubernetes.pod.namespace': default
  ingress:
  - fromEndpoints:
    - matchLabels:
        'k8s:io.kubernetes.pod.namespace': kube-system
    toPorts:
    - ports:
      - port: "8080"
        protocol: TCP
```
